app/models/affiliate.rb

```diff
@@ -0,0 +1,8 @@
+class Affiliate < ApplicationRecord
+
+    has_many :coordinators
+    has_many: tutors
+
+    has_many :matches
```

Comment:
Sounds like a good validation when setting up a match. The match should only be valid (this would be a controller level validation) if the student and the tutor have the same affiliate. 

It might be possible on the match model, too.